bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -390,27 +395,32 @@ void PreProcessor::finalizePreProcessing(NodeIdType clientId) {
   const auto &clientEntry = ongoingRequests_[clientId];
   {
     lock_guard<mutex> lock(clientEntry->mutex);
-    reqSeqNum = clientEntry->reqProcessingStatePtr->getReqSeqNum();
-    // Copy of the message body is unavoidable here, as we need to create a new message type which lifetime is
-    // controlled by the replica while all PreProcessReply messages get released here.
-    clientRequestMsg =
-        make_unique<ClientRequestMsg>(clientId,
-                                      HAS_PRE_PROCESSED_FLAG,
-                                      reqSeqNum,
-                                      clientEntry->reqProcessingStatePtr->getPrimaryPreProcessedResultLen(),
-                                      clientEntry->reqProcessingStatePtr->getPrimaryPreProcessedResult(),
-                                      clientEntry->reqProcessingStatePtr->getReqTimeoutMilli(),
-                                      clientEntry->reqProcessingStatePtr->getReqCid());
+    auto &reqProcessingStatePtr = clientEntry->reqProcessingStatePtr;
+    if (reqProcessingStatePtr) {
+      reqSeqNum = reqProcessingStatePtr->getReqSeqNum();
+      // Copy of the message body is unavoidable here, as we need to create a new message type which lifetime is
+      // controlled by the replica while all PreProcessReply messages get released here.
+      clientRequestMsg = make_unique<ClientRequestMsg>(clientId,
+                                                       HAS_PRE_PROCESSED_FLAG,
+                                                       reqSeqNum,
+                                                       reqProcessingStatePtr->getPrimaryPreProcessedResultLen(),
+                                                       reqProcessingStatePtr->getPrimaryPreProcessedResult(),
+                                                       reqProcessingStatePtr->getReqTimeoutMilli(),
+                                                       reqProcessingStatePtr->getReqCid());
+      incomingMsgsStorage_->pushExternalMsg(move(clientRequestMsg));
+      preProcessorMetrics_.preProcReqSentForFurtherProcessing.Get().Inc();
+      releaseClientPreProcessRequest(clientEntry, clientId);
+      LOG_INFO(GL, "Pre-processing completed for clientId=" << clientId << " reqSeqNum=" << reqSeqNum);
```

Comment:
LOG_DEBUG, to avoid flooding the logs?

Reply:
I would like to leave the start/end pre-execution command logs for now. Will make it 'DEBUG' later.